pkg/operator/nfs/controller.go

```diff
@@ -0,0 +1,480 @@
+					Ports: []v1.ContainerPort{
+						{
+							Name:          "nfs-port",
+							ContainerPort: int32(nfsPort),
+						},
+						{
+							Name:          "rpc-port",
+							ContainerPort: int32(rpcPort),
+						},
+					},
+					VolumeMounts: createVolumeMountList(&nfsServer.spec),
+					SecurityContext: &v1.SecurityContext{
+						Capabilities: &v1.Capabilities{
+							Add: []v1.Capability{
+								"SYS_ADMIN",
+								"DAC_READ_SEARCH",
+							},
+						},
+					},
+				},
+			},
+			Volumes: createPVCSpecList(&nfsServer.spec),
+		},
+	}
+
+	return nfsPodSpec
+}
+
+func (c *Controller) createNfsStatefulSet(nfsServer *nfsServer, replicas int32) error {
+	appsClient := c.context.Clientset.AppsV1beta2()
+
+	nfsPodSpec := c.createNfsPodSpec(nfsServer)
+
+	statefulSet := v1beta2.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            nfsServer.name,
+			Namespace:       nfsServer.namespace,
+			Labels:          createAppLabels(),
+			OwnerReferences: []metav1.OwnerReference{nfsServer.ownerRef},
+		},
+		Spec: v1beta2.StatefulSetSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: createAppLabels(),
+			},
+			Template:    nfsPodSpec,
+			ServiceName: nfsServer.name,
+		},
+	}
+
+	if _, err := appsClient.StatefulSets(nfsServer.namespace).Create(&statefulSet); err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return err
+		}
+		logger.Infof("stateful set %s already exists in namespace %s", statefulSet.Name, statefulSet.Namespace)
+	} else {
+		logger.Infof("stateful set %s created in namespace %s", statefulSet.Name, statefulSet.Namespace)
+	}
+
+	return nil
+}
+
+func (c *Controller) onAdd(obj interface{}) {
+	nfsObj := obj.(*nfsv1alpha1.NFSServer).DeepCopy()
+
+	nfsServer := newNfsServer(nfsObj, c.context)
+
+	logger.Infof("new NFS server %s added to namespace %s", nfsObj.Name, nfsServer.namespace)
+
+	logger.Infof("validating nfs server server spec in namespace %s", nfsServer.namespace)
+	if err := validateServerConfig(&nfsServer.spec); err != nil {
+		logger.Errorf("Invalid NFS server configuration spec %+v", err)
+	}
+
+	logger.Infof("creating nfs server service in namespace %s", nfsServer.namespace)
+	if err := c.createNFSService(nfsServer); err != nil {
+		logger.Errorf("Unable to create NFS service %+v", err)
+	}
+
+	logger.Infof("creating nfs server configuration in namespace %s", nfsServer.namespace)
+	if err := c.createNFSConfigMap(nfsServer); err != nil {
+		logger.Errorf("Unable to create NFS ConfigMap %+v", err)
+	}
+
+	logger.Infof("creating nfs server stateful set in namespace %s", nfsServer.namespace)
+	if err := c.createNfsStatefulSet(nfsServer, int32(nfsServer.spec.Replicas)); err != nil {
+		logger.Errorf("Unable to create NFS stateful set %+v", err)
+	}
+}
+
+func (c *Controller) onUpdate(oldObj, newObj interface{}) {
+	oldNfsServ := oldObj.(*nfsv1alpha1.NFSServer).DeepCopy()
+	newNfsServ := newObj.(*nfsv1alpha1.NFSServer).DeepCopy()
+
+	_ = oldNfsServ
```

Comment:
instead of referencing the unused variables just to make the compiler happy, i'd suggest deleting the unused code